l2rpn_baselines/utils/BaseDeepQ.py

```diff
@@ -213,14 +234,19 @@ def load_network(self, path, name=None, ext="h5"):
         if self.verbose:
             print("Succesfully loaded network.")
 
-    def target_train(self):
+    def target_train(self, tau=None):
         """
         update the target model with the parameters given in the :attr:`BaseDeepQ._training_param`.
         """
-        # nothing has changed from the original implementation
+        if tau is None:
+            tau = self._training_param.tau
+
```

Comment:
get_weights also returns non-trainable params, thus updating some params that do not need to.
Also, tau inv can be computed only once. 

```python
def update_target_soft(source_model, target_model, tau=1e-3):
        tau_inv = 1.0 - tau
        # Get parameters to update
        target_params = target_model.trainable_variables
        source_params = source_model.trainable_variables

        # Update each param
        for src, dest in zip(source_params, target_params):
            # Polyak averaging
            var_update = src.value() * tau
            var_persist = dest.value() * tau_inv
            dest.assign(var_update + var_persist)
```